default/python/AI/AIstate.py

```diff
@@ -158,12 +165,15 @@ def __setstate__(self, state):
                 convert_to_version(state, v+1)
             self.__dict__ = state
         except ConversionError:
-            try:
-                aggression = state['character'].get_trait(Aggression).key
-            except Exception as e:
-                print >> sys.stderr, "Could not find the aggression level of the AI, defaulting to typical."
-                print >> sys.stderr, e
-                aggression = fo.aggression.typical
+            if '_aggression' in state:
+                aggression = state['_aggression']
+            else:
+                try:
```

Comment:
If an ConversionError was caught, then the savegame could not be converted (e.g. ill-defined conversion function, invalid version number, invalid pickle-data...).

Considering we couldn't "load" the AIstate, we need to create a new one. That requires an aggression level which we try to extract from the saved state. If that fails, we then default to the typical aggression level.

Probably needs an explanatory comment. Any other issues with the approach?